dipy/workflows/io.py

```diff
@@ -112,6 +118,44 @@ def run(self, input_files,
                 logging.info('Total number of non-unit bvectors {0}\n'
                              .format(ncl1))
 
+            if extension in ['.trk', '.tck', '.trx', '.vtk', '.vtp', '.fib',
+                             '.dpy']:
+
+                sft = None
+                if extension in ['.trk', '.trx']:
+                    sft = load_tractogram(input_path, 'same',
+                                          bbox_valid_check=False)
+                else:
+                    sft = load_tractogram(input_path, reference,
+                                          bbox_valid_check=False)
+
+                lengths = [len(s) for s in sft.streamlines]
+                lengths_mm = list(length(sft.streamlines))
```

Comment:
I think this line gives you the number of streamlines, not their lengths in mm

Reply:
I confirm that it is the lengths in mm. See below the result:
```
❯ dipy_info test_fast_tracking/tractogram_c_disco.trk
INFO:----------------------------------------------------
INFO:Looking at test_fast_tracking/tractogram_c_disco.trk
INFO:----------------------------------------------------
INFO:Number of streamlines: 33750
INFO:min_length_mm: 0.19999771700283706
INFO:mean_length_mm: 9.685641917422362
INFO:max_length_mm: 77.10583827216257
INFO:std_length_mm: 13.186300696375488
INFO:min_length_nb_points: 2.0
INFO:mean_length_nb_points: 44.88044444444444
INFO:max_length_nb_points: 363.0
INFO:std_length_nb_points: 62.540960585152355
INFO:min_step_size: 0.19999562203884125
INFO:mean_step_size: 0.22072798013687134
INFO:max_step_size: 26.52972984313965
INFO:std_step_size: 0.2722674012184143
INFO:data_per_point_keys: []
INFO:data_per_streamline_keys: []
```

Reply:
That confuses me. the `streamlines` attribute of a StatefulTractogram object is either a list or an ArraySequence. If it's a list, how could `len(sft.streamlines)` possibly know anything about the lengths of the individual streamlines? And if it's an ArraySequence the same question (see https://github.com/dipy/dipy/blob/master/dipy/io/stateful_tractogram.py#L228-L230).

Reply:
Not sure to understand your question.

this `length(sft.streamlines)` and not `len(sft.streamlines)`.  Are you confusing both? (homemade function vs standard function).  You can see the documentation of `length`  here: https://github.com/dipy/dipy/blob/24f76c8eb30aa9d256b2e163987139fd0c9bda68/dipy/tracking/streamlinespeed.pyx#L55.  I hope it helps

Reply:
Yes - that's exactly what I was confused about.